src/components/input/Select.tsx

```diff
@@ -203,25 +209,32 @@ function SelectOption<T>({
           />
         )}
         {multiple && (
-          <div
-            className={classnames('scale-125', {
-              'text-grey-6': selected,
-              'text-grey-3 hover:text-grey-6': !selected,
-            })}
-          >
-            <Checkbox
-              checked={selected}
-              checkedIcon={CheckboxCheckedFilledIcon}
-              elementRef={checkboxRef}
-              onChange={toggleValue}
-              onKeyDown={e => {
-                if (e.key === 'ArrowLeft') {
-                  e.preventDefault();
-                  optionRef.current?.focus();
-                }
-              }}
-            />
-          </div>
+          <Checkbox
+            containerClasses={classnames(
+              // Add negative margin, padding and invisible border to match the
+              // container's, preventing accidental miss-clicks outside the
+              // checkbox that mess the whole selection.
+              '-m-2.5 p-2.5 border border-transparent',
+              // The checkbox is sized based on the container's font size. Make
+              // it a bit larger.
+              'text-xl',
```

Comment:
The checkbox was previously enlarged via `scale-125`, but that conflicts now with the margin and padding added here.

Increasing the font size was always the right approach.